sycl-fusion/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -0,0 +1,116 @@
+//==-------------------------- KernelFusion.cpp ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "KernelFusion.h"
+#include "Kernel.h"
+#include "KernelIO.h"
+#include "Options.h"
+#include "fusion/FusionHelper.h"
+#include "helper/ConfigHelper.h"
+#include "helper/ErrorHandling.h"
+#include "translation/SPIRVLLVMTranslation.h"
+#include <llvm/Support/Error.h>
+#include <sstream>
+
+using namespace jit_compiler;
+
+using FusedFunction = helper::FusionHelper::FusedFunction;
+using FusedFunctionList = std::vector<FusedFunction>;
+
+static FusionResult errorToFusionResult(llvm::Error &&Err,
+                                        const std::string &Msg) {
+  std::stringstream ErrMsg;
+  ErrMsg << Msg << "\nDetailed information:\n";
+  llvm::handleAllErrors(std::move(Err),
+                        [&ErrMsg](const llvm::StringError &Err) {
+                          // Cannot throw an exception here if LLVM itself is
+                          // compiled without exception support.
+                          ErrMsg << "\t" << Err.getMessage() << "\n";
```

Comment:
Nit; I am not a big fan of shadowing `Err` here. Granted it is fully valid, it can be confusing if you don't notice the lambda argument shadowing it. It's a very small function though, so it's by no means blocking.

Reply:
Renamed to avoid shadowing.